shared/src/commonMain/kotlin/com/example/travelapp_kmp/listing/ListScreenViewModel.kt

```diff
@@ -54,37 +56,52 @@ class ListScreenViewModel {
     }
     fun onAction(actions: ListViewModelActions) {
         viewModelScope.launch {
+
             when (actions) {
                 is ListViewModelActions.OnCountrySelected -> {
-                    emitNewState(actions)
+                    emitNewState(
+                        actions,
+                        countriesApi.getWeather(actions.country.touristPlaces[0].location)
+                    )
                 }
 
                 is ListViewModelActions.OnItemSwiped -> {
-                    state.emit(
-                        (state.value as ListScreenState.Success).copy(
-                            selectedItemIndex = actions.index
-                        )
-                    )
+                    onChangePlaceSelected(actions.index)
                 }
 
                 is ListViewModelActions.MoveToIndex -> {
-                    val previousState = (state.value as ListScreenState.Success)
-                    state.emit(
-                        previousState.copy(
-                            selectedItemIndex = actions.index,
-                        )
-                    )
+                    onChangePlaceSelected(actions.index)
                 }
             }
         }
     }
 
-    private suspend fun emitNewState(actions: ListViewModelActions.OnCountrySelected) {
+    private suspend fun onChangePlaceSelected(index: Int) {
+        with(state.value as ListScreenState.Success) {
+            selectedCountry.touristPlaces[selectedItemIndex].location.let {
+                countriesApi.getWeather(it)
+            }.let { weather ->
+                state.emit(
+                    this.copy(
+                        selectedItemIndex = index,
+                        weatherSelectedCountry = weather
+                    )
+                )
+            }
+
+        }
+    }
+
```

Comment:
I am wondering about this code.
So whenever we change the location, it will try to fetch the weather data. So it might take time. Wont the user will be stuck?

Reply:
If we can someone make it independent of each other like the weather data is fetched async.